### DISCUSSION TITLE
npm package : importLibrary does not update the user lib

### DISCUSSION THREAD

**Discussion by pomdtr** [excalidraw/excalidraw#4023] [Q&A]
Here is the code I use to test it:

```
 excalidrawRef.current!.importLibrary("https://libraries.excalidraw.com/libraries/thebrahmnicboy/Logic-Gates.excalidrawlib")
```

However it does not have any effect on the drawing.
I do not see any error either.

**Comment by pomdtr** [comment]
Ok i just tested it in the codesandbox, it looks like the alert is not showing from the vscode webview. I'll try to disable it using the csrf token.

**Reply by pomdtr** [reply]
Ok this is a vscode limitation. See https://github.com/Microsoft/vscode/issues/67109

**Comment by pomdtr** [comment]
Ok it looks like I need to:

- set the libraryReturnUrl prop to a vscode url scheme
- Wait for the user to click on the browse libraries items
- parse the csrf token from the callback url
- call the importLibrary helper using both the the url param and the csrf token

Does it sounds right ?

**Reply by dwelle** [reply]
Yep. I'm thinking we should have simplified the API to `importLibraryFromUrl(url, prompt = true)`, alas.
